ci/tests/disklayout/disk_roles_test.py

```diff
@@ -0,0 +1,256 @@
+                if not number_of_roles_to_remain == 0:
+                    if len(partition.roles) < number_of_roles_to_remain:
+                        logger.warning("Number of roles that should remain exceed the number of roles that are present! Keeping all roles instead!")
+                        roles_list = partition.roles
+                    else:
+                        roles_list = partition.roles[number_of_roles_to_remain:]
+                    remaining_roles = General.remove_list_from_list(partition.roles, roles_list)
+                else:
+                    # When number_of_roles_to_remain, everything should have been removed
+                    remaining_roles = []
+                logger.info("Removing roles '{0}' from partition '{1}'".format(remaining_roles, partition.guid))
+                GeneralDisk.adjust_disk_role(partition, remaining_roles, 'SET')
+                # Will test if the role is an empty list
+                collection[partition.guid] = remaining_roles
+        return collection
+        # End remove disk roles
+
+    @staticmethod
+    def validate_roles(collection):
+        """
+
+        :param collection: object containing the partition guid as key and the list of roles as value
+        :type collection: Object
+
+        :return: returns true or false based on the condition
+        :type: boolean
+        """
+        # Start validation
+        # Check if roles are the same as specified
+        iterations = 0
+        succesfull_iterations = 0
+        logger.info("Starting validation of disk roles")
+        for key, value in collection.iteritems():
+            iterations += 1
+            # Fetch partition matching key
+            partition = GeneralDisk.get_disk_partition(key)
+            # Check if roles are the same
+            logger.info("Comparing roles on the partition '{0}'...".format(key))
+            logger.info("Found '{0}' on partition and predefined roles: '{1}'".format(partition.roles, value))
+            if sorted(partition.roles) == sorted(value):
+                succesfull_iterations += 1
+            else:
+                logger.error("The role '{0}' for partition '{1}' was not set correctly!".format(value, key))
+                logger.error("Found '{0}' and expected '{1}' was not set correctly!".format(partition.roles, value))
+        return succesfull_iterations == iterations
+
+    def tdr_0001_add_remove_role_and_crosscheck_model_test(self, ip, configuration):
+        """
+        This test will add a DB role to the sda disk of the storage router with the given IP
+        :param ip: IP address of a storage router. (Example:
+        :type ip: str
+
+        :param configuration: Dict that determines layout
+        :type configuration: dict
+        :return: None
+        """
+        # Start input validation
+        pattern = re.compile(r"^(?<!\S)((\d|[1-9]\d|1\d\d|2[0-4]\d|25[0-5])\b|\.\b){7}(?!\S)$")
```

Comment:
IP validation could be useful for other tests too, so I'd suggest we create a general_network.py in ci/tests/general and add this as a static method there ...